cmd/soroban-cli/src/commands/keys/add.rs

```diff
@@ -31,9 +43,32 @@ pub struct Cmd {
 impl Cmd {
     pub fn run(&self, global_args: &global::Args) -> Result<(), Error> {
         let print = Print::new(global_args.quiet);
-        let secret = self.secrets.read_secret()?;
+        let secret = self.read_secret(&print)?;
         let path = self.config_locator.write_identity(&self.name, &secret)?;
         print.checkln(format!("Key saved with alias {:?} in {path:?}", self.name));
         Ok(())
     }
+
+    fn read_secret(&self, print: &Print) -> Result<Secret, Error> {
```

Comment:
moved this here from `config/secret.rs` - i was seeing an interesting cyclical dependency, and moving this here helped that be a bit less confusing.